runtime/src/bank.rs

```diff
@@ -105,18 +106,26 @@ pub struct Bank {
 
     /// Slot leader
     leader: Pubkey,
+
+    pub id: u64,
 }
 
 impl Bank {
     pub fn new(genesis_block: &GenesisBlock) -> Self {
+        Self::new_with_paths(&genesis_block, "")
```

Comment:
Will that make a mess of the current directory? Cause conflicts in parallel tests?

Reply:
No, it has a default of `$OUT_DIR/accountsdb/<random-pubkey>` if it is the empty string.

Reply:
Okay, let's make that None instead of empty string then. And resolve that within `new_with_paths`. And then have `new` also take an Option, but in that one, None means use a HashMap.